phases/19-capstone-projects/52-experiment-runner/code/main.py

```
@@ -0,0 +1,300 @@
+            "knob": self.knob,
+            "rows": [{"value": v, "result": r.to_dict()} for v, r in self.rows],
+        }
+
+
+class AblationRunner:
+    def __init__(self, runner: ExperimentRunner) -> None:
+        self._runner = runner
+
+    def sweep(self, base: ExperimentSpec, knob: str, values: Iterable[Any]) -> AblationTable:
+        rows: list[tuple[Any, ExperimentResult]] = []
+        for value, spec in zip(values, ablate(base, knob, values)):
```

Ruff (0.15.14)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)